tests/priority_queue_test.c

```diff
@@ -41,32 +42,42 @@ static int test_priority_queue_preserves_order(struct aws_allocator *alloc, void
     ASSERT_INT_EQUALS(5, num_elements, "Priority queue size should have been %d but was %d", 5, num_elements);
 
     int pop_val, top_val;
-    aws_priority_queue_top(&queue, &top_val);
-    aws_priority_queue_pop(&queue, &pop_val);
+    err = aws_priority_queue_top(&queue, &top_val);
```

Comment:
NIT, I see why you think this was a pain in the ass now:

All you had to do was:
ASSERT_SUCCESS(aws_priority_queue_dynamic_init(&queue, alloc, 10, sizeof(int), int_compare), "Priority queue initialization failed with error %d", err);

This is fine though.

Reply:
If I had done that, the error message wouldn't include the error code. Which makes it less useful.
Anyways, vim makes that kind of stuff easy.